virtual_rainforest/models/animals/animal_cohorts.py

```diff
@@ -105,55 +108,98 @@ def metabolize(self, dt: timedelta64) -> None:
+            )
+            * self.individuals
+        )
         # Minimum of the energy available and amount that can be consumed in an 8 hour
         # foraging window .
-        food.energy -= consumed_energy * self.individuals
+        plant_food.energy -= consumed_energy
         # The amount of energy consumed by the average member * number of individuals.
         self.stored_energy += (
-            consumed_energy * self.functional_group.conversion_efficiency
+            consumed_energy
+            * self.functional_group.conversion_efficiency
+            * self.functional_group.mechanical_efficiency
         )
         # The energy [J] extracted from the PlantCommunity adjusted for energetic
         # conversion efficiency and divided by the number of individuals in the cohort.
+        soil.energy += consumed_energy * (
+            1 - self.functional_group.mechanical_efficiency
+        )
+
         return consumed_energy
 
-    def excrete(self, soil: PalatableSoil, consumed_energy: float) -> None:
-        """The function to transfer waste energy from an animal cohort to the soil.
+    def predation(self, prey: "AnimalCohort", carcass_pool: CarcassPool) -> float:
+        """A predation event where this cohort preys on the given prey cohort.
 
         Args:
-            soil: The local PalatableSoil pool in which waste is deposited.
-            consumed_energy: The amount of energy flowing through cohort digestion.
+            prey: The AnimalCohort instance being preyed upon.
+            carcass_pool: The resident pool of animal carcasses to which the remains of
+              dead individuals are delivered.
+
+        Returns:
+            A float containing the amount of energy consumed in the herbivory bout.
 
         """
-        waste_energy = consumed_energy * self.functional_group.conversion_efficiency
-        soil.energy += waste_energy * self.individuals
-        # The amount of waste by the average cohort member * number individuals.
+        # Calculate the number of individuals that can be eaten based on intake rate
+        # Here we assume predator can consume prey mass equivalent to its daily intake
+        number_eaten = int(min(self.intake_rate / prey.mass, prey.individuals))
+
+        # Calculate the energy gain from eating prey
+        # Here we assume all eaten mass is converted to energy
+        consumed_energy = min(
+            (
+                number_eaten
+                * prey.mass
+                * ENERGY_DENSITY["meat"]
+                * self.functional_group.mechanical_efficiency
+            ),
+            prey.stored_energy,
+        )
 
-    def forage(self, food: PlantCommunity, soil: PalatableSoil) -> None:
-        """The function to enact multi-step foraging behaviors.
+        # Reduce the number of individuals in the prey cohort and add energy to predator
+        prey.individuals -= number_eaten
+        carcass_pool.energy += consumed_energy * (
```

Comment:
If the explanation is about adding energy to predator, why is it added to the carcass pool?